src/setuptools_pyproject_migration/_long_description.py

```diff
@@ -0,0 +1,260 @@
+            return LongDescriptionMetadata(text, content_type="text/plain")
+
+    def pyproject_readme(self) -> Union[str, ReadmeFile, ReadmeText]:
+        """
+        Compute the value that should be added to the ``pyproject.readme`` entry
+        in ``pyproject.toml`` to represent this long description.
+
+        >>> LongDescriptionMetadata("Description", "text/plain", pathlib.Path("readme.txt")).pyproject_readme()
+        {'file': 'readme.txt', 'content-type': 'text/plain'}
+        >>> LongDescriptionMetadata("Description", "text/plain", None).pyproject_readme()
+        {'text': 'Description', 'content-type': 'text/plain'}
+
+        This class does not make any assumptions about the content type, so it
+        will not emit a content type if one was not provided, but in that case
+        the filename needs to have an extension so that consumers will be able to
+        infer it from that extension.
+
+        >>> LongDescriptionMetadata("Description", None, pathlib.Path("readme.txt")).pyproject_readme()
+        'readme.txt'
+        """
+        if self.content_type:
+            if self.path:
+                return {"file": str(self.path), "content-type": self.content_type}
+            else:
+                return {"text": self.text, "content-type": self.content_type}
+        else:
+            # We should never get to a situation where there is no content type
+            # stored in this object unless the content type can be inferred from
+            # the filename.
+            if not (self.path and self.path.suffixes):
+                raise ValueError(f"No content type provided or can be inferred from filename {self.path}")
+            return str(self.path)
+
+
+def _raw_long_description_value(dist: setuptools.dist.Distribution) -> str:
+    """
+    Return the raw value of the ``long_description`` option given to setuptools.
+
+    This will be either the value in ``setup.cfg``, or the value passed to
+    the ``setup()`` function.
+    """
+    try:
+        # If this is present, it contains the raw text value of the
+        # long_description option from setup.cfg. (Or, it should. Otherwise we'd
+        # have to manually parse setup.cfg to get it.) If it names a file,
+        # get_long_description() should give the string obtained by
+        # reading the file.
+        return dist.command_options["metadata"]["long_description"][1]
+    except KeyError:
+        # If the option comes from setup.py then get_long_description()
+        # gives us the raw value instead. We can't necessarily get the
+        # filename this way because setup() might never receive the
+        # filename if the file was read directly by setup.py.
+        return dist.get_long_description()
+
+
+def _read_long_description_file(path: pathlib.Path) -> str:
+    """
+    Read the content of the long description from a filename.
+    """
+    # TODO handle multiple comma-separated filenames
```

Comment:
I'll just put an issue here so we don't forget about this later… no rush of course.

https://github.com/diazona/setuptools-pyproject-migration/issues/158